### ISSUE TITLE
[BUG]: ImperfectSplittingWarning rounding

### ISSUE THREAD

**Issue by KnathanM** [JacksonBurns/astartes#175]
**Describe the bug**

For a "dataset" of length 10 with splits 0.5, 0.25, 0.25, astartes tells me the validation has 0.30 of the data when it really has 0.2 of the data. 10 isn't a real dataset size, but I'm using it for a demonstration of the splitting functions in Chemprop. 

**Example(s)**
```
from astartes import train_val_test_split
train, val, test = train_val_test_split([0,1,2,3,4,5,6,7,8,9], sampler="random", train_size=0.5, val_size=0.25, test_size=0.25)
```
Outputs:
".../astartes/main.py:325: ImperfectSplittingWarning: Actual train/test split differs from requested size. Requested validation size of 0.25, got 0.30. Requested test size of 0.25, got 0.30."

But val has length 2 not 3. test does have length 3.

**Expected behavior**

".../astartes/main.py:325: ImperfectSplittingWarning: Actual train/test split differs from requested size. Requested validation size of 0.25, got 0.20. Requested test size of 0.25, got 0.30."

**Environment**

- python version 3.11.8
# Name                    Version                   Build  Channel
astartes                  1.2.2                    pypi_0    pypi
- OS Linux

**Checklist**
- [ ] all dependencies are satisifed: `conda list` shows the packages listed in the `README`
- [ ] the unit tests are working: `pytest -v` reports no errors

Haven't checked these, but can if that is helpful

**Comment by JacksonBurns** [comment]
Thanks for the report! I suspect the rounding is off. I will look into this!

**Comment by JacksonBurns** [comment]
Turns out it was even simpler, I was printing out the sizes for the test set instead of the validation set:

https://github.com/JacksonBurns/astartes/blob/09aa54bbba7063ff8a94f1dfa8f71adc7c07688f/astartes/main.py#L314-L318

Resolved in the linked PR!
